src/components/shared/ScrollToFieldError/index.js

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react';
+
+export const ScrollToFieldError = ({
+  scrollBehavior = { left: 0, behavior: 'smooth' },
+  useFormikContext,
+  fieldsOrder = [],
+}) => {
+  const { submitCount, isValid, errors } = useFormikContext();
+
+  useEffect(() => {
+    if (isValid) {
+      return;
+    }
+
+    const fieldErrorNames = Object.keys(errors);
+
+    const firstFieldErrorName = fieldsOrder.find((fieldName) =>
+      fieldErrorNames.includes(fieldName),
+    );
+    if (!firstFieldErrorName) {
+      return;
+    }
+
+    const element = document.querySelector(`input[name='${firstFieldErrorName}']`);
+    if (!element) {
+      return;
+    }
```

Comment:
I am not sure if it is better or not, but I think that you could identify the first error from the DOM in place of using `fieldsOrder`? Am I missing something?

Reply:
I had not thought of that, but it's a good idea. I'm going to do that